Kvrocks failed to build in macos-13 CI workflow

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.


### Version

unstable

### Minimal reproduce step

Refer to https://github.com/apache/kvrocks/actions/runs/14351177410/job/40294270737?pr=2867.

```
[  0%] Building CXX object _deps/cpptrace-build/CMakeFiles/cpptrace-lib.dir/src/binary/elf.cpp.o
compiling static library
clang: warning: no such sysroot directory: '-DXXH_NAMESPACE=LZ4_' [-Wmissing-sysroot]
lz4.c:224:11: fatal error: 'stdlib.h' file not found
# include <stdlib.h>   /* malloc, calloc, free */
[  0%] Building CXX object _deps/cpptrace-build/CMakeFiles/cpptrace-lib.dir/src/binary/mach-o.cpp.o
          ^~~~~~~~~~
CC obj/conf_bf260fcd81b3c57e99d10f6ae607c7ea/static/debug.o
1 error generated.
CC obj/conf_bf260fcd81b3c57e99d10f6ae607c7ea/static/entropy_common.o
lz4file.c:34:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>  /* malloc, free */
[  0%] Building CXX object _deps/cpptrace-build/CMakeFiles/cpptrace-lib.dir/src/binary/module_base.cpp.o
         ^~~~~~~~~~
1 error generated.
lz4frame.c:89:10: fatal error: 'string.h' file not found
#include <string.h>   /* memset, memcpy, memmove */
         ^~~~~~~~~~
1 error generated.
In file included from lz4hc.c:67:
./lz4.c:224:11: fatal error: 'stdlib.h' file not found
# include <stdlib.h>   /* malloc, calloc, free */
          ^~~~~~~~~~
1 error generated.
xxhash.c:107:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
make[4]: *** [liblz4.a] Error 1
make[3]: *** [CMakeFiles/make_lz4] Error 2
make[2]: *** [CMakeFiles/make_lz4.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```

### What did you expect to see?

--

### What did you see instead?

--

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/.github/workflows/kvrocks.yaml
+++ b/.github/workflows/kvrocks.yaml
@@ -109,24 +109,37 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - name: Darwin Clang
-            os: macos-13
-            compiler: auto
           - name: Darwin Clang arm64
             os: macos-14
             compiler: auto
-          - name: Darwin Clang without Jemalloc
-            os: macos-13
+          - name: Darwin Clang arm64 without Jemalloc
+            os: macos-14
             compiler: auto
             without_jemalloc: -DDISABLE_JEMALLOC=ON
-          - name: Darwin Clang with OpenSSL
-            os: macos-13
+          - name: Darwin Clang arm64 with OpenSSL
+            os: macos-14
             compiler: auto
             with_openssl: -DENABLE_OPENSSL=ON
-          - name: Darwin Clang without luaJIT
-            os: macos-13
+          - name: Darwin Clang arm64 without luaJIT
+            os: macos-14
             compiler: auto
             without_luajit: -DENABLE_LUAJIT=OFF
+          # FIXME: temporarily disable macos-13 build, refer to #2874
+          # - name: Darwin Clang
+          #   os: macos-13
+          #   compiler: auto
+          # - name: Darwin Clang without Jemalloc
+          #   os: macos-13
+          #   compiler: auto
+          #   without_jemalloc: -DDISABLE_JEMALLOC=ON
+          # - name: Darwin Clang with OpenSSL
+          #   os: macos-13
+          #   compiler: auto
+          #   with_openssl: -DENABLE_OPENSSL=ON
+          # - name: Darwin Clang without luaJIT
+          #   os: macos-13
+          #   compiler: auto
+          #   without_luajit: -DENABLE_LUAJIT=OFF
           - name: Ubuntu GCC
             os: ubuntu-22.04
             compiler: gcc

